### ISSUE TITLE
Only mark form as needing redeployment if form contents changes 

### ISSUE THREAD

**Issue by pmusaraj** [kobotoolbox/kpi#1690]
Any change to the project's settings (including name, description, or other details) marks the form as requiring to be redeployed -- even though the form itself hasn't changed. 

Expected behavior: Unless the form itself is changed, don't prompt the user to redeploy.

Supersedes #1469.

**Comment by pmusaraj** [comment]
Feel free to have a look at this Leszek, it's a little tricky but may be interesting.

**Comment by pmusaraj** [comment]
See also https://github.com/kobotoolbox/kpi/issues/443

**Comment by smutnyleszek** [comment]
I will read the code and see how it works :)

**Comment by alnf** [comment]
It would be nice to have this implemented, since the message about redeploy is very misleading.

**Comment by jnm** [comment]
`<name>` is actually included in the XForm generated by KC, so to get a name change to appear in Collect or Enekto, a redeployment _is_ required. Changes in the other metadata have no bearing on the deployed XForm, so the message should be suppressed for them.

If we decide what we want to consider redeployment-worthy content, then we could hash that and have the backend report the hash in the API representation. The frontend could compare the latest content hash (instead of version ID) to the deployed content hash and display the redeployment message only if the hashes differ. I'd suggest including `AssetVersion.name` and `AssetVersion.version_content` in the hash.

There's also an `AssetVersion.deployed_content`, but it seems mostly unused:
```
>>> AssetVersion.objects.exclude(deployed_content=None).order_by('date_modified').first().date_modified
datetime.datetime(2016, 10, 3, 6, 41, 44, 4000, tzinfo=<UTC>)
>>> AssetVersion.objects.exclude(deployed_content=None).order_by('date_modified').last().date_modified
datetime.datetime(2016, 10, 18, 6, 52, 40, 822997, tzinfo=<UTC>)
>>> AssetVersion.objects.exclude(deployed_content=None).count()
42
>>> AssetVersion.objects.count()
1039699
```

**Comment by pmusaraj** [comment]
Comparing hashes sounds good to me. And I agree with including `AssetVersion.name` and `AssetVersion.version_content`.

**Comment by jnm** [comment]
Blocked by https://github.com/kobotoolbox/kpi/issues/1833

**Comment by magicznyleszek** [comment]
@jnm I see that #1833 is closed - does it mean it's not blocked anymore?

**Comment by magicznyleszek** [comment]
Waiting for #1941 and frontend work can start

**Comment by noliveleger** [comment]
@magicznyleszek: PR #1941 is now closed.

**Comment by jnm** [comment]
Well, that was nice, but as Leszek [mentioned](), it's not useful without knowing the content hash of the current version. I'll add `version__content_hash` to the asset detail view, as an analogue of `version_id`.
